usine.py

```diff
@@ -1,23 +1,349 @@
+            return Config(self.value.get(key))
+        except AttributeError:
+            return Config()
+
+    def __setattr__(self, key, value):
+        self.value[key] = value
+
+    def __str__(self):
+        return str(self.value)
+
+    def __eq__(self, other):
+        return other == self.value
+
+    # https://eev.ee/blog/2012/03/24/python-faq-equality/
+    # No way to override "is" behaviour, so no way to do "config.key is None"…
+    def __bool__(self):
+        return bool(self.value)
+
+    def __hash__(self):
+        return hash(self.value)
+
+    def __iter__(self):
+        return iter(self.value)
+
+    def items(self):
+        return self.value.items()
+
+    def update(self, other):
+        self.value.update(other)
+
+
+config = Config()  # singleton.
+
+
+class Template(string.Template):
+    # Default delimiter ($) clashes at least with Nginx DSL.
+    delimiter = '$$'
+
+
+def template(path, **context):
+    with Path(path).open() as f:
+        template = Template(f.read())
+        return StringIO(template.substitute(**context))
+
+
+class Formatter(string.Formatter):
+    """
+    Allow to have some custom formatting types.
+
+    B: boolean attribute
+    S: small boolean attribute
+    V: k=v like attribute
+    """
+
+    def _vformat(self, format_string, args, kwargs, used_args, recursion_depth,
+                 auto_arg_index=0):
+        result = []
+        for literal_text, name, spec, conversion in self.parse(format_string):
+
+            # output the literal text
+            if literal_text:
+                result.append(literal_text)
+
+            # if there's a field, output it
+            if name:
+                # given the name, find the object it references
+                #  and the argument it came from
+                obj, _ = self.get_field(name, args, kwargs)
+                obj = self.convert_field(obj, conversion)
+
+                value = ''
+                if spec == 'bool':
+                    if obj:
+                        value = '--' + name.replace('_', '-')
+                elif spec == 'initial':
+                    if obj:
+                        value = '-' + name[0]
+                elif spec == 'equal':
+                    if obj:
+                        value = '--' + name.replace('_', '-') + '=' + str(obj)
+                else:
+                    value = self.format_field(obj, spec)
+
+                # format the object and append to the result
+                result.append(value)
+
+        return ''.join(result), auto_arg_index
+
+
+def formattable(func):
+
+    def wrapper(*args, **kwargs):
+        spec = inspect.signature(func)
+        for idx, (name, param) in enumerate(spec.parameters.items()):
+            if idx < len(args):
+                client.context[name] = args[idx]
+            else:
+                client.context[name] = kwargs.get(name, param.default)
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+class Status:
```

Comment:
Shall we use/inherit from [subprocess.CompletedProcess](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess) instead?